File: client/android/utils/src/main/java/com/yandex/div/core/Disposable.java
package com.yandex.div.core;

import java.io.Closeable;

/**
 * It supposed to be a {@link AutoCloseable} but, we have to support up to API 16.
 *
 * In comparison with {@link Closeable} IOException is dropped from the {@link #close()} method.
 */
public interface Disposable extends Closeable {

    Disposable NULL = () -> { };

    /**
     * Closes this resource, relinquishing any underlying resources.
     * <p>
     * Implementation of this method must be idempotent.
     */
    void close();
}

File: client/android/div-core/src/main/java/com/yandex/div/core/Disposable.kt
package com.yandex.div.core

import java.io.Closeable

/**
 * An object that may hold resources (such as file or socket handles) until it is closed.
 * In comparison with [Closeable] IOException is dropped from the [close] method.
 */
public fun interface Disposable : AutoCloseable, Closeable {
    /**
     * Closes this resource, relinquishing any underlying resources.
     *
     * Implementation of this method must be idempotent.
     */
    override fun close()

    public companion object {

        @JvmField
        public val NULL: Disposable = Disposable { }
    }
}
